src/libraries/System.Private.CoreLib/src/System/Diagnostics/Contracts/Contracts.cs

```diff
@@ -628,7 +628,8 @@ private static void AssertMustUseRewriter(ContractFailureKind kind, string contr
             Assembly? probablyNotRewritten = null;
             for (int i = 0; i < stack.FrameCount; i++)
             {
-                Assembly? caller = stack.GetFrame(i)!.GetMethod()?.DeclaringType!.Assembly;
+                MethodBase stackMethod = stack.GetFrame(i)!.GetMethod();
+                Assembly? caller = (stackMethod != null) ? stackMethod.DeclaringType!.Assembly : null;
```

Comment:
DeclaringType can be null. I do not think there is anything guaranteeing that it will be non-null here. This is one of those cases nullable annotations caught a real bug.

Should the fix be to just change `!` after DeclaringType to `?`? No need to split it into multiple lines.

Reply:
We try to avoid any product changes during nullability annotations as those are always risky and can get lost in a large PR. In this case since PR is only affecting this specific change so I think we should fix it.